pkg/operator/controllers/dnsmasq/machineconfigpool_controller_test.go

```diff
@@ -0,0 +1,254 @@
+					Spec: arov1alpha1.ClusterSpec{
+						OperatorFlags: arov1alpha1.OperatorFlags{
+							controllerEnabled: "true",
+						},
+					},
+				},
+			),
+			mcocli: fakeMco(
+				&mcv1.MachineConfigPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "custom",
+						DeletionTimestamp: &metav1.Time{Time: time.Unix(0, 0)},
+					},
+					Status: mcv1.MachineConfigPoolStatus{},
+					Spec:   mcv1.MachineConfigPoolSpec{},
+				},
+			),
+			mocks: func(mdh *mock_dynamichelper.MockInterface) {},
+			request: ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "",
+					Name:      "custom",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "MachineConfigPool marked for deletion with controller finalizer deletes ARO DNS MachineConfig and removes finalizer",
+			arocli: fakeAro(
+				&arov1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Status:     arov1alpha1.ClusterStatus{},
+					Spec: arov1alpha1.ClusterSpec{
+						OperatorFlags: arov1alpha1.OperatorFlags{
+							controllerEnabled: "true",
+						},
+					},
+				},
+			),
+			mcocli: fakeMco(
+				&mcv1.MachineConfigPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:              "custom",
+						DeletionTimestamp: &metav1.Time{Time: time.Unix(0, 0)},
+						Finalizers:        []string{MachineConfigPoolControllerName},
+					},
+					Status: mcv1.MachineConfigPoolStatus{},
+					Spec:   mcv1.MachineConfigPoolSpec{},
+				},
+			),
+			mocks: func(mdh *mock_dynamichelper.MockInterface) {
+				mdh.EXPECT().EnsureDeleted(gomock.Any(), "MachineConfig", "", "99-custom-aro-dns").Times(1)
+				mdh.EXPECT().Ensure(gomock.Any(), gomock.AssignableToTypeOf(&mcv1.MachineConfigPool{})).Times(1)
+			},
+			request: ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "",
+					Name:      "custom",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "MachineConfigPool reconciles ARO DNS MachineConfig",
+			arocli: fakeAro(
+				&arov1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Status:     arov1alpha1.ClusterStatus{},
+					Spec: arov1alpha1.ClusterSpec{
+						OperatorFlags: arov1alpha1.OperatorFlags{
+							controllerEnabled: "true",
+						},
+					},
+				},
+			),
+			mcocli: fakeMco(
+				&mcv1.MachineConfigPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:       "custom",
+						Finalizers: []string{MachineConfigPoolControllerName},
+					},
+					Status: mcv1.MachineConfigPoolStatus{},
+					Spec:   mcv1.MachineConfigPoolSpec{},
+				},
+			),
+			mocks: func(mdh *mock_dynamichelper.MockInterface) {
+				mdh.EXPECT().Ensure(gomock.Any(), gomock.AssignableToTypeOf(&mcv1.MachineConfig{})).Times(1)
+			},
+			request: ctrl.Request{
+				NamespacedName: types.NamespacedName{
+					Namespace: "",
+					Name:      "custom",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "MachineConfigPool with no finalizer adds finalizer",
```

Comment:
maybe we can also test that the finalizer is not readded after the mcp has been marked for deletion and the finalizer was run

Reply:
The `MachineConfigPool marked for deletion with no finalizers does nothing` case validates this by ensuring the MCP is not modified.